code/datums/components/telegraph_ability.dm

```diff
@@ -0,0 +1,47 @@
+/**
+ * Component given to creatures to telegraph their abilities!
+ */
+/datum/component/basic_mob_ability_telegraph
+	/// how long before we use our attack
+	var/telegraph_time
+	/// sound to play, if any
+	var/sound_path
+	/// are we currently telegraphing
+	var/currently_telegraphing = FALSE
+
+/datum/component/basic_mob_ability_telegraph/Initialize(telegraph_time = 1 SECONDS, sound_path)
+
+	if(!isliving(parent))
+		return COMPONENT_INCOMPATIBLE
+	src.telegraph_time = telegraph_time
+	src.sound_path = sound_path
+
+/datum/component/basic_mob_ability_telegraph/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_MOB_ABILITY_STARTED, PROC_REF(on_ability_activate))
+
+/datum/component/basic_mob_ability_telegraph/UnregisterFromParent()
+	UnregisterSignal(parent, COMSIG_MOB_ABILITY_STARTED)
+
+/datum/component/basic_mob_ability_telegraph/proc/on_ability_activate(mob/living/source, datum/action/cooldown/activated, atom/target)
+	SIGNAL_HANDLER
+
+	if(currently_telegraphing)
+		return COMPONENT_BLOCK_ABILITY_START
+
+	if(!activated.IsAvailable())
+		return
+
+	currently_telegraphing = !currently_telegraphing
```

Comment:
```suggestion
	currently_telegraphing = FALSE
```

it's never going to be anything else at this point in the code, let's choose to be explicit